### PR TITLE
Grant access to chrome.sockets APIs for IPFS companion

### PR DESCRIPTION
**Pull request by bbondy**
Fix https://github.com/brave/brave-browser/issues/3876

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [ ]
  - [ ] Linux
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [x] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: common/extensions/api/_api_features.json [hunk 1]
```diff
@@ -2,7 +2,8 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 // Command to get whitelist ID: echo -n mnojpmjdmbbfmejpflffifhffcmidifd | openssl sha1 | tr '[:lower:]' '[:upper:]'
-// Same for sync api:           echo -n nomlkjnggnifocmealianaaiobmebgil | openssl sha1
+// Same for sync api:           echo -n nomlkjnggnifocmealianaaiobmebgil | openssl sha1 | tr '[:lower:]' '[:upper:]'
+// Same for ipfs-companion:     echo -n nibjojkomfdiaoajekhjakgkdhaomnch | openssl sha1 | tr '[:lower:]' '[:upper:]'
```
Comment (diracdeltas) [inline common/extensions/api/_api_features.json]: `780bf954c0f7c586ea9662d4f967771f49cc2114` is the sha1 of `nibjojkomfdiaoajekhjakgkdhaomnch`; what does `FF32507DC3DB5DFFD1D6733187C84D4B74713D63` correspond to?

File: common/extensions/api/_api_features.json [hunk 2]
```diff
@@ -61,17 +62,17 @@
   "sockets.tcp": {
     "dependencies": ["manifest:sockets"],
     "contexts": ["blessed_extension"],
-    "whitelist": ["3D9518A72EB02667A773B69DBA9E72E0F4A37423"]
```
Comment (diracdeltas) [inline common/extensions/api/_api_features.json]: i assume 3D9518A72EB02667A773B69DBA9E72E0F4A37423 is webtorrent?

Comment (bbondy) [reply common/extensions/api/_api_features.json]: yep

File: common/extensions/api/_api_features.json (hunk 1 above)
Comment (bbondy) [reply common/extensions/api/_api_features.json]: IPFS beta extension, it's the 2nd commit in the PR.

**Comment by bbondy** [comment]
@diracdeltas I cleaned up the headers to specify the mapping and ID for each extension.

**Pull request merged**